### DISCUSSION TITLE
PDF++ et Zotero

### DISCUSSION THREAD

**Discussion by bepolymathe** [RyotaUshio/obsidian-pdf-plus#122] [Q&A]
Good morning, 

 Thank you really for this wonderful plugin.
 I use Zotero for managing my bibliography and there are some very good community plugins for that.  But one thing has always generated friction for me.  I annotate in Zotero and I have to export these annotations into Obsidian with more or less sophisticated templates... and the problems that this poses when I then add annotations in Zotero or Obsidian.  The arrival of PDF++ could change that. 

 Have you already experienced interesting workflows with Zotero and PDF++?

**Comment by maxmilneaus** [comment]
Hello there @bepolymathe, 

I am here for this discussion also. Currently I highlight in Zotero and then transition across to Obsidian. This plugin is allowing me to do all of the work in Zotero and thus allows a very slick system where I can click on an page link in Obisidna and it takes me directly to that place on the PDF. All links update when changes are made and etc. What I am thinking about, and wondering if it would be more fun to have a FaceTime about it, is what is the most conducive work flow for Zotero and Obsidian. 

I'm studying as Masters in Counselling and have a back ground in Advertising with a focus on systems design. 

If up for it. DM me. Then we could even offer back a work flow suggestion for the benefit of others.

**Comment by RyotaUshio** [comment]
> I use Zotero for managing my bibliography and there are some very good community plugins for that. But one thing has always generated friction for me. I annotate in Zotero and I have to export these annotations into Obsidian with more or less sophisticated templates... and the problems that this poses when I then add annotations in Zotero or Obsidian. The arrival of PDF++ could change that.

Thanks, I was feeling exactly the same frustration with Zotero-related plugins and that was why I started writing this plugin. Glad it also worked out for you.

My Zotero workflow is very simple: I use Zotero exclusively for scraping bibliographic metadata (using the Chrome extension), and then I use the Zotero Integration plugin to import it to Obsidian. After that, I don't interact with Zotero and I do all my work in Obsidian using PDF++.
I have a symbolic link to Zotero's attachments folder `~/Zotero/storage` in my vault, so any PDFs saved by Zotero can be accessed from within Obsidian (see also: #38).

This approach increases the size of your vault, which can be problematic if you're using Obsidian Sync.
Alternatively, you can use [PDF++'s feature for opening PDFs on the web](https://ryotaushio.github.io/obsidian-pdf-plus/external-pdf-files) to avoid importing actual PDF files into the vault.


But I'm no expert in Zotero and I guess there are a lot of things that I don't know about Zotero, which might help me make my workflow even better. If there are any interesting features of Zotero, I'm eager to know. I'm looking forward to seeing comments from you and other users in this thread:)

**Reply by KelvinHYYip** [reply]
@RyotaUshio Is there a plan to add metadata scraping to PDF++? It will be very handy and make it a self-sufficient app. Anyway, well done Ryota! I enjoy using the plug-in very much. ;)

**Reply by RyotaUshio** [reply]
Thanks! I think PDF++ would never be a better metadata scraper than Zotero is even if I were to ever give it a try, so probably I won't. Trying to be good at everything is in general not a great idea imo.

**Comment by maxmilneaus** [comment]
> > I use Zotero for managing my bibliography and there are some very good community plugins for that. But one thing has always generated friction for me. I annotate in Zotero and I have to export these annotations into Obsidian with more or less sophisticated templates... and the problems that this poses when I then add annotations in Zotero or Obsidian. The arrival of PDF++ could change that.

Interesting this is exactly what I want to do

> Thanks, I was feeling exactly the same frustration with Zotero-related plugins and that was why I started writing this plugin. Glad it also worked out for you.

You are a supreme legend and I thank you for that.

> My Zotero workflow is very simple: I use Zotero exclusively for scraping bibliographic metadata (using the Chrome extension), and then I use the Zotero Integration plugin to import it to Obsidian. After that, I don't interact with Zotero and I do all my work in Obsidian using PDF++. I have a symbolic link to Zotero's attachments folder `~/Zotero/storage` in my vault, so any PDFs saved by Zotero can be accessed from within Obsidian (see also: #38).

@RyotaUshio I would love to see a screen share of your Zotero import process. 

When I pull up my Zotero PDF folder, it's just a random bunch of numbers. Is that what you're symbolic linking to/from? Or is it more organised?

At the moment, I have my whole vault syncing via iCloud, so I usually just import the PDF directly into obsidian so it's all in one spot. Using your process it means I wouldn't be able to access a PDF if I were to open my vault on an iPad, because the PDFs would be under desktop Zotero folder. Correct?

> This approach increases the size of your vault, which can be problematic if you're using Obsidian Sync. Alternatively, you can use [PDF++'s feature for opening PDFs on the web](https://ryotaushio.github.io/obsidian-pdf-plus/external-pdf-files) to avoid importing actual PDF files into the vault.

Curious, my main goal is to have everything is one vault package where I don't double up on document duplicates, this allows for easy pick up and go in case of redundancy and also making backups.  

Thanks again for your work and I'm really interested to see how you developed this and what obsidian end up picking up on and implementing themselves. I appreciate your values of dedication and diversity in the way in which this plug-in can be used.

**Reply by maxmilneaus** [reply]
Attached below is the template I use to link Zotero to Obsidian. It is a combination of [Elena Razlogova](https://publish.obsidian.md/history-notes/00+About+the+Project) and [Alexandra Phelan ](https://medium.com/@alexandraphelan/an-updated-academic-workflow-zotero-obsidian-cffef080addd) work. 

[Zotero Template  - Max.md](https://github.com/RyotaUshio/obsidian-pdf-plus/files/14767196/Zotero.Template.-.Max.md)

**Reply by RyotaUshio** [reply]
> When I pull up my Zotero PDF folder, it's just a random bunch of numbers

I forgot to mention I'm using the ZotFile addon. It automatically renames attachments so that their file names are human-readable, like `[author]_[year]_[title].pdf`.

> Is that what you're symbolic linking to/from? Or is it more organised?

My symbolic link is pointing to `~/Zotero/storage` (I'm on macOS), and the symbolic link is located in the root folder of my vault.
I name it `_Zotero`. As a result, my vault's structure looks like this:

```
[VAULT ROOT]/
├─ _Notes/
├─ _Zotero/
   ├─ AVD83W7F/
   │  ├─ [author]_[year]_[title].pdf
   ├─ JFAS8FF3S/
      ├─ [author]_[year]_[title].pdf
```
(The names of the folders that the Zotero PDFs live in are still strange random strings, which is a little bit annoying)

> At the moment, I have my whole vault syncing via iCloud, so I usually just import the PDF directly into obsidian so it's all in one spot. Using your process it means I wouldn't be able to access a PDF if I were to open my vault on an iPad, because the PDFs would be under desktop Zotero folder. Correct?

I haven't tried it but I guess so. But if you are using Obsidian Sync, it syncs PDFs under the symlinked folder as well, and you can read it on iPad, too.

Also thanks for sharing your template! My template (and some additional info) can be found here: https://github.com/RyotaUshio/obsidian-pdf-plus/wiki/Example-workflow:-Zotero-integration

**Comment by bepolymathe** [comment]
Hello everyone, 

Thanks for your feedback. Personally, for importing a list I use a bit the same system as @RyotaUshio but with the Zotlit plugin which is faster in use. On the other hand, I prefer to leave all PDFs in Zotero because I sometimes annotate them in Zotero, especially since there are mobile/tablet versions. So my problem is that I have both annotations taken in Zotero and others in Obsidian. I'm still trying to figure out how it works...

**Reply by maxmilneaus** [reply]
I am same same when it comes to using Zotero on iPad/Mobile and find it helpful for annotating there. Interested to hear and track what you all do in future.

**Reply by JhCandiceYao** [reply]
This is a bit off-topic, but I’m curious—how do both of you use Zotero on mobile for reading and annotating? I always find it frustrating because of the PDF size on phone screen makes everything small and how difficult it is to navigate or highlight accurately.

**Comment by alberti42** [comment]
Related to the discussion, I created a plugin to use an external BibTeX file which manages 10000 PDF documents in my library. I could integrate my plugin with PDF++. It works very nicely for me. I understand this is not the solution for Zotero. However, I opened a discussion to collect ideas of what could be a generalized solution that allows such extensions in the future.

To make everything work for me, currently, I am having to fork PDF++ and change a couple of lines of code. It would be great if developers would have a way to link to PDF++ to provide extensions for managing external libraries such as BibTeX (in my case) or Zotero in your case.

**Reply by bepolymathe** [reply]
It's very interesting. Zotero can easily create a bibtex file, so that might help. @RyotaUshio ?

**Reply by alberti42** [reply]
> It's very interesting. Zotero can easily create a bibtex file, so that might help. @RyotaUshio ?

I am confident we could extend my plugin to also interface with Zotero and thus integrate with PDF++. Is it something you are looking for? I don't know much the situation of what is currently possible in terms of integration between PDF++ and Zotero.

PS: I am not a Zotero user because I have been working for many years with BibTex+BibDesk and have many scripts existing for importing and exporting papers into my library automatically.

**Reply by alberti42** [reply]
I am not sure why but the original link of my plugin disappeared from the conversation above. I post here again in case you have missed: https://github.com/alberti42/obsidian-bibtex-integration

**Comment by JayamanideepR** [comment]
Really cool suggestions to integrate zotero and obsidian!

I would like to share my workflow which is similar to previous suggestions with slight variations.
Similar to some previous suggestions, I also use Zotero just for managing citations and read my papers and take notes directly in obsidian (using pdf++). 
I use pdf++, citations and templater plugins (in obsidian) in combination with better bibtex plugin (in Zotero) for this workflow. 

A big thank you to all these plugin and software contributors for making such awesome tools!

Here is my workflow:
1. Place entire Zotero folder inside the obsidian vault
	1. This is a crude way of doing it but others might have better suggestions on how to link zotero files in obsidian vault
	2. I primarily do this so that I can open my pdf files and follow similar workflow in my other synced devices
2. In Zotero:
	1. Using better bibtex plugin, export all the papers into a bibtex file and choose the option to keep it constantly updated (so that you don't need to manually update it each time you add a new paper)
3. In Obsidian:
	1. Use the [citations plugin](https://github.com/hans/obsidian-citation-plugin) to parse the bibtex file (this plugin can help you in creating and/or opening a literature note for each citation in the bibtex file)
		1. Make sure to choose the bibtex file exported from Zotero as citation database path
4. Use the template attached at the end of this message in the citation plugin, for creating a new note (**needs templater plugin**)
5. Once you create a new literature note using citation plugin, you can now see the pdf file link in the callout at the top. You can just click on it to open the file with pdf++ plugin and make your annotations into this literature note
6. Whenever you want to recover your literature note, use `Citations: Open literature note` or `ctrl+shift+o` and search for the reference name

```
---
title: "{{title}}"
authors: "{{authorString}}"
year: {{year}}
pdf_file_path: {{entry.data.fields.file}}
tags: 
- zot_notes
---

>[!info] Pdf file
>[[<%tp.user.get_file_name(String.raw `${tp.frontmatter.pdf_file_path}`)%>]]

## Reading notes
 <% tp.file.cursor(1) %>
```

\*Sometimes, the file field in bibtex file contains a html link to the webpage the paper is saved from, In that case, I just check the metadata of the note and copy the file name from the `pdf_file_path` field

**Reply by maxmilneaus** [reply]
Thank you so much for your workflow process. I'm going to give it a go on my end and will let you know if I have additional thoughts.

**Comment by callumalpass** [comment]
I've written a [plugin](https://github.com/callumalpass/obsidian-biblib) which I think pairs nicely with PDF++, and helps avoid complicated syncing with Zotero. It might be of interest for you who use both Zotero and PDF++. 

BibLib is very simple, but it allows you to manage your literature notes and references all from within Obsidian. It does this by querying Citoid (which itself uses Zotero translation servers in the backend) to pull bibliographic information when given a URL, ISBN, or DOI, and then converts this to CSL-JSON. This CSL data is then added as YAML frontmatter to a literature note. 

It also provides a command for scraping all your literature notes to build either a CSL bibliography file, or a Bibtex file. 

I've used this system of keeping my reference data in literature note frontmatter for a few years, but only just recently polished a plugin to help automate it in Obsidian. 

PDF++ has been super helpful for this workflow--my favourite Obsidian plugin by far!

**Reply by bepolymathe** [reply]
Very interesting @callumalpass! Thanks.

**Reply by bepolymathe** [reply]
I'll wait until it's available in the Obsidian store to install it. But this project is very promising and not restrictive... 👍🏼

**Reply by maxmilneaus** [reply]
I don't fully understand the Coding Languages and how this plugin works - though based upon the Gif in your plugin it all makes sense and would see me pretty much bypassing Zotero. Except for in one instance. The PDF highlighting in Zotero on iPad and iPhone is great. I'm hoping PDF++ can come up with something to make that process easier then I would be ALL IN on your plugin. Thank you for sharing and the development energy you have put in and being willing to generously share it.

[![](https://raw.githubusercontent.com/callumalpass/obsidian-biblib/refs/heads/main/screenshots/create-lit-note.gif)](url)

**Reply by maxmilneaus** [reply]
Alright. It is looking likely that I can use Obsidian as my dedicated PDF annotated and reader on iOS and iPadOS - so I am going to give yours a go. 

For those interested here is my work flow on iPad highlighting:

## PDF++ Copy Text Format Setup
My recommended settings:
- Custom Copy Format "Text": `{{selection}} {{linkWithDisplay}}`

This places the page number link after your selected text. Example:
> "Mainstream psychiatry follows this tradition. Over the past decade the Departments of Defense and Veterans Affairs combined have spent over $4.5 billion on antidepressants, antipsychotics, and antianxiety drugs." p.186

(The markdown behind this: `[[Van der Kolk - 2015 - The body keeps the score mind, brain and body in the transformation of trauma.pdf#page=186&selection=12,0,14,38|p.186]]`)

## Setting Up Your Highlights Note
1. Create a new note with your preferred properties and formatting. I use a system that imports Zotero properties into Obsidian.

## Optimal iPad PDF++ Settings
1. Enable Auto Copy
2. Enable Auto Paste
   - Set Target Markdown File to: "Last active & open md if any, otherwise last pasted"
   - Disable "Focus Editor After Auto-Pasting"
   - Disable "Respect Current Cursor Position"
   - Enable "Blank line above the appended content"
3. Under General
   - Disable "Open target markdown file is not opened" (crucial for preventing the file from opening after each highlight)

## Starting Your Highlighting Workflow
1. Open your highlights destination note in a new tab
2. Verify Auto Highlight and Auto Paste are enabled (look for the text icon and clipboard with arrow icons in the ribbon)
3. Highlight your first text selection
4. Check the destination tab to confirm it pasted correctly
5. Have fun, get smart, share ideas and be kindly cheeky!

Hope this helps! I'll make a video on this in the future, but this is the simplest approach for now.

**Reply by bepolymathe** [reply]
Thanks for your feedback. Looking forward to seeing your video.

**Comment by maxmilneaus** [comment]
Hello Candice,Yes. The screen is too small on iOS I use iPad exclusively for the “mobile” version. On 2 May 2025, at 01:23, Candice Yao ***@***.***> wrote:﻿
This is a bit off-topic, but I’m curious—how do both of you use Zotero on mobile for reading and annotating? I always find it frustrating because of the PDF size on phone screen makes everything small and how difficult it is to navigate or highlight accurately.
